In [1]:
import pandas as pd
import numpy as np

In [21]:
Data = pd.read_csv(r"C:\Users\Cypher\Desktop\Data Science and ML\GitHub\Data-Scientist-Salaries\ds_salaries.csv")
df = Data

### Let's get an overview of the Data

In [22]:
df.head()

Unnamed: 0  work_year experience_level employment_type  \
0           0       2020               MI              FT   
1           1       2020               SE              FT   
2           2       2020               SE              FT   
3           3       2020               MI              FT   
4           4       2020               SE              FT   

                    job_title  salary salary_currency  salary_in_usd  \
0              Data Scientist   70000             EUR          79833   
1  Machine Learning Scientist  260000             USD         260000   
2           Big Data Engineer   85000             GBP         109024   
3        Product Data Analyst   20000             USD          20000   
4   Machine Learning Engineer  150000             USD         150000   

  employee_residence  remote_ratio company_location company_size  
0                 DE             0               DE            L  
1                 JP             0               JP            S  
2                 GB            50               GB            M  
3                 HN             0               HN            S  
4                 US            50               US            L

In [23]:
df.columns

Index(['Unnamed: 0', 'work_year', 'experience_level', 'employment_type',
       'job_title', 'salary', 'salary_currency', 'salary_in_usd',
       'employee_residence', 'remote_ratio', 'company_location',
       'company_size'],
      dtype='object')

In [24]:
df.shape

(607, 12)

In [25]:
df.describe()

Unnamed: 0    work_year        salary  salary_in_usd  remote_ratio
count  607.000000   607.000000  6.070000e+02     607.000000     607.00000
mean   303.000000  2021.405272  3.240001e+05  112297.869852      70.92257
std    175.370085     0.692133  1.544357e+06   70957.259411      40.70913
min      0.000000  2020.000000  4.000000e+03    2859.000000       0.00000
25%    151.500000  2021.000000  7.000000e+04   62726.000000      50.00000
50%    303.000000  2022.000000  1.150000e+05  101570.000000     100.00000
75%    454.500000  2022.000000  1.650000e+05  150000.000000     100.00000
max    606.000000  2022.000000  3.040000e+07  600000.000000     100.00000

In [26]:
df.isnull().sum()

Unnamed: 0            0
work_year             0
experience_level      0
employment_type       0
job_title             0
salary                0
salary_currency       0
salary_in_usd         0
employee_residence    0
remote_ratio          0
company_location      0
company_size          0
dtype: int64

The Data has 607 Rows and 11 Columns, It has no missing values. 
It Requires little Data Clean Up; Apply Seniority ranking

### Data Cleaning

In [31]:
df["senior"] = df.job_title.apply(lambda x: 1 if "lead" in x.lower() or "senior" in x.lower() or "head" in x.lower() or
                               "principal" in x.lower() or "director" in x.lower() else 0)
df.senior.value_counts()

0    562
1     45
Name: job_title, dtype: int64

In [32]:
df.head()

Unnamed: 0  work_year experience_level employment_type  \
0           0       2020               MI              FT   
1           1       2020               SE              FT   
2           2       2020               SE              FT   
3           3       2020               MI              FT   
4           4       2020               SE              FT   

                    job_title  salary salary_currency  salary_in_usd  \
0              Data Scientist   70000             EUR          79833   
1  Machine Learning Scientist  260000             USD         260000   
2           Big Data Engineer   85000             GBP         109024   
3        Product Data Analyst   20000             USD          20000   
4   Machine Learning Engineer  150000             USD         150000   

  employee_residence  remote_ratio company_location company_size  senior  
0                 DE             0               DE            L       0  
1                 JP             0               JP            S       0  
2                 GB            50               GB            M       0  
3                 HN             0               HN            S       0  
4                 US            50               US            L       0

In [33]:
df.shape

(607, 13)

In [36]:
df.drop("salary", axis=1, inplace = True)

In [38]:
df.drop("Unnamed: 0", axis=1, inplace = True)

In [39]:
df

work_year experience_level employment_type                   job_title  \
0         2020               MI              FT              Data Scientist   
1         2020               SE              FT  Machine Learning Scientist   
2         2020               SE              FT           Big Data Engineer   
3         2020               MI              FT        Product Data Analyst   
4         2020               SE              FT   Machine Learning Engineer   
..         ...              ...             ...                         ...   
602       2022               SE              FT               Data Engineer   
603       2022               SE              FT               Data Engineer   
604       2022               SE              FT                Data Analyst   
605       2022               SE              FT                Data Analyst   
606       2022               MI              FT                AI Scientist   

    salary_currency  salary_in_usd employee_residence  remote_ratio  \
0               EUR          79833                 DE             0   
1               USD         260000                 JP             0   
2               GBP         109024                 GB            50   
3               USD          20000                 HN             0   
4               USD         150000                 US            50   
..              ...            ...                ...           ...   
602             USD         154000                 US           100   
603             USD         126000                 US           100   
604             USD         129000                 US             0   
605             USD         150000                 US           100   
606             USD         200000                 IN           100   

    company_location company_size  senior  
0                 DE            L       0  
1                 JP            S       0  
2                 GB            M       0  
3                 HN            S       0  
4                 US            L       0  
..               ...          ...     ...  
602               US            M       0  
603               US            M       0  
604               US            M       0  
605               US            M       0  
606               US            L       0  

[607 rows x 11 columns]

In [44]:
df.to_csv(r"C:\Users\Cypher\Desktop\Data Science and ML\GitHub\Data-Scientist-Salaries\data_cleaned.csv",index = False)

AttributeError: module 'pandas' has no attribute 'to_csv'